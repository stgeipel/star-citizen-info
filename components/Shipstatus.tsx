import { IStatus } from '../types/Ship'

const Shipstatus = ({ status }: { status: IStatus }) => {
  if (status.value == 'Flight Ready') {
    return (
      <span className="inline-flex items-center px-3 py-0.5 rounded-full text-sm font-medium leading-5 bg-green-100 text-green-800">
        {status.value}
      </span>
    )
  }
  return (
    <span className="inline-flex items-center px-3 py-0.5 rounded-full text-sm font-medium leading-5 bg-red-100 text-red-800">
      {status.value}
    </span>
  )
}

export default Shipstatus
