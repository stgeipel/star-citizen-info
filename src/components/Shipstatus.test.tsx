import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Shipstatus from './Shipstatus';

describe('Shipstatus', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(
      <Shipstatus status={{ value: 'Flight Ready' }}></Shipstatus>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
