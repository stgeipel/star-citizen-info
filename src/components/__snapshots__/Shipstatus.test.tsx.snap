// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shipstatus matches snapshot 1`] = `
<span
  className="inline-flex items-center px-3 py-0.5 rounded-full text-sm font-medium leading-5 bg-green-100 text-green-800"
>
  Flight Ready
</span>
`;
